test/basictests.jl

```diff
@@ -121,3 +123,167 @@ for u0 in [1.0, [1, 1.0]]
     @test solve(probN, NewtonRaphson()).u ≈ sol
     @test solve(probN, NewtonRaphson(; autodiff = false)).u ≈ sol
 end
+
+# --- TrustRegion tests ---
+
+function benchmark_immutable(f, u0)
+    probN = NonlinearProblem(f, u0)
+    solver = init(probN, TrustRegion(), abstol = 1e-9)
+    sol = solve!(solver)
+end
+
+function benchmark_mutable(f, u0)
+    probN = NonlinearProblem(f, u0)
+    solver = init(probN, TrustRegion(), abstol = 1e-9)
+    sol = solve!(solver)
```

Comment:
Can you add a test that `solve!` on the mutable code path does not allocate? I think if that passes then it's done.